src/validation.h

```diff
@@ -907,7 +914,7 @@ class ChainstateManager
     //                                  constructor
     //! @param[in] snapshot_blockhash   If given, signify that this chainstate
     //!                                 is based on a snapshot.
-    CChainState& InitializeChainstate(CTxMemPool& mempool, const std::optional<uint256>& snapshot_blockhash = std::nullopt)
+    CChainState& InitializeChainstate(CTxMemPool* mempool, const std::optional<uint256>& snapshot_blockhash = std::nullopt)
```

Comment:
What do you think about making this std::optional<CTxMemPool, std::optional>?

In this case, snapshot_blockhash can't be possibly specified without mempool, and that's what I originally suggested.

Reply:
There are cases when we want to call this with a snapshot blockhash but not a mempool - during snapshot load time when we need a chainstate instance to populate/validate but don't want to yet make it the active chainstate in case loading the UTXO snapshot fails.